docs/usage/writing_edsl.rst

```diff
@@ -32,29 +33,14 @@ sum over axis `0` of a 2D tensor (in Keras this would be ``K.sum(I, axis=0)``):
 
    .. group-tab:: C++
 
-        .. code-block:: cpp
-
-          Tensor sum_over_axis(const Tensor& I) {
-            TensorDim M, N;
-            TensorIndex m, n;
-            I.bind_dims(M, N);
-            auto O = TensorOutput(N);
-            O(n) += I(m, n); // contraction
-            return O;
-          }
+      .. literalinclude:: ../codeexamples/writing_edsl.cc
+        :language: cpp
+        :lines: 27-34
 
    .. group-tab:: Python
 
-        .. code-block:: python
-
-          def sum_over_axis(I):
-            M, N = TensorDims(2)
-            m, n = TensorIndexes(2)
-            I.bind_dims(M, N)
-            O = TensorOutput(N)
-            # contraction
-            O[n] += I[m, n]
-            return O
+      .. literalinclude:: ../codeexamples/writing_edsl.py
```

Comment:
Since we're including entire functions, can we use the built-ins that exist for the `literalinclude` directive? This will make it _much_ easier to maintain because we won't have to specify line numbers everywhere.

> The directive also supports including only parts of the file. If it is a Python module, you can select a class, function or method to include using the pyobject option:

```
.. literalinclude:: example.py
   :pyobject: Timer.start
```

[Read more...](https://www.sphinx-doc.org/en/1.5/markup/code.html#directive-literalinclude)

Reply:
I think this only works for python, but i'm +100 on doing this for python :D

Reply:
It might not be much better than line numbers, but there are `start-after` and `end-before` directives for `literalinclude` that will at least let you use regexes to include lines of code rather than line numbers...

```
.. literalinclude:: ShowingCodeExamplesInSphinx.rst
   :language: rst
   :start-after: Instead of using
   :end-before: For example
```

[Read more...](https://build-me-the-docs-please.readthedocs.io/en/latest/Using_Sphinx/ShowingCodeExamplesInSphinx.html#literalinclude-directive)